### PR TITLE
[UI] Applications: Adding a default image for the broken / unsupported application Icons.

### LINKED ISSUE #6161: [UI] Applications: Helm logo missing / broken link
#### Current Behavior
Screen Shot 2022-09-01 at 3 04 52 PM

#### Desired Behavior
Display Helm logo.

#### Environment

- Browser: Safari
- Host OS: Mac 
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by kamalbuilds**
**Description**

A ss taken from the current situation.
image

Addition of a default image renders this -> 
image

This PR fixes #6161  

**Notes for Reviewers**

If there's no application type defined or one defined, but that is not supported (e.g. " "), we are trying to  incorporate a default icon which is taken to be https://drive.google.com/drive/folders/1izNTsbspq554e0qxCzv6d3AvjvIUkg4D as all other icons are also colored.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by Chadha93 — COMMENTED** [review]
@legendarykamal please add the new fix ss too. You can search the previous application or upload an undefined app from server. Although unsupported app can be uploaded from UI.

**Comment by legendarykamal** [comment]
> @legendarykamal eslint check is failing here. 😬. Are you adding no verify flag while committing?

No , There is 1 error generated , the rest are key-prop warnings not related to this PR . I will fix the error. 👍

**Comment by Chadha93** [comment]
@legendarykamal Oh, I missed typed here we don't allow uploading an unsupported type in the application.

**Comment by legendarykamal** [comment]
@Chadha93 fixed the ui eslint error and updated the required ss.

File: ui/components/MesheryApplications/ApplicationsCard.js [hunk 1]
```diff
@@ -83,7 +83,10 @@ function MesheryApplicationCard({
                   e.stopPropagation(); handleAppDownload(id ,source_type, name)
                 }}
               >
-                <img src={`/static/img/${(source_type).replaceAll(" ", "_").toLowerCase()}.svg`} width="45px" height="45px" />
+                {source_type== undefined || source_type == " " ?
+                  <img src={`/static/img/default_apps-green.svg`} width="45px" height="45px" />
```
Comment (Chadha93) [inline ui/components/MesheryApplications/ApplicationsCard.js:87]: @legendarykamal Please update this condition in the table view also.

Comment (legendarykamal) [reply ui/components/MesheryApplications/ApplicationsCard.js:87]: oh yes, needful done  ✅ @Chadha93

**Comment by Chadha93** [comment]
@legendarykamal Eslint workflow is fixed in master. :heavy_check_mark: 
Will you look into the eslint errors on your PR? Please update the PR.

File: ui/components/MesheryApplications.js [hunk 2]
```diff
@@ -648,7 +648,10 @@ function MesheryApplications({
                 title="click to download"
                 onClick={() => handleAppDownload(rowData.id ,rowData.type.String, rowData.name)}
               >
-                <img src={`/static/img/${(rowData.type.String).replaceAll(" ", "_").toLowerCase()}.svg`} width="45px" height="45px" />
+                {source_type== undefined || source_type == " " ?
```
Comment (Chadha93) [inline ui/components/MesheryApplications.js:651]: Uh Oh!  @legendarykamal This is showing that `source_type` is not defined. I'm now wondering how it worked for you locally. Did you test your changes?
image

Comment (legendarykamal) [reply ui/components/MesheryApplications.js:651]: I see 👀 actually i had tested for grid view(ss is from it) and not table.
 This arises from the tabel view I will fix this right away .

Comment (legendarykamal) [reply ui/components/MesheryApplications.js:651]: My bad I had seen and was gonna fix it earlier itself 😓 .

File: ui/components/MesheryApplications/ApplicationsCard.js [hunk 3]
```diff
@@ -83,7 +83,10 @@ function MesheryApplicationCard({
...
+                  : <img src={`/static/img/${(source_type).replaceAll(" ", "_").toLowerCase()}.svg`} width="45px" height="45px" />
```
Comment (leecalcote) [inline ui/components/MesheryApplications/ApplicationsCard.js:88]: Static assignment of both height and width is not advised. Minimally, please only have one specified.

It'd be great to see flex used here and across the rest of Meshery UI's components.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: ui/components/MesheryApplications/ApplicationsCard.js (hunk 3 above)
Comment (leecalcote) [reply ui/components/MesheryApplications/ApplicationsCard.js:88]: Unaddressed feedback remains.

Comment (Chadha93) [reply ui/components/MesheryApplications/ApplicationsCard.js:88]: // @legendarykamal

Comment (legendarykamal) [reply ui/components/MesheryApplications/ApplicationsCard.js:88]: on it .

**Comment by leecalcote** [comment]
https://github.com/meshery/meshery/issues/6161#issuecomment-1379372765

**Comment by leecalcote** [comment]
Thanks for your help, @legendarykamal. Let's close out this PR as no longer desired.

**Pull request closed without merging**